pkg/cli/cli.go

```diff
@@ -1232,6 +1244,126 @@ func runConnectionWebrtcAgentCommand(flags *ConnectFlags, c *command, args []str
 	return nil
 }
 
+func runConnectionWebSocketAgentCommand(flags *ConnectFlags, c *command, args []string, opts *subCommandOpts) error {
+	// Change Web URL to WebSocket URL
+	// e.g. https://127.0.0.1:1443/ -> wss://127.0.0.1:1443
+	wsRoot := flags.ServiceURL
+	wsRoot = strings.ReplaceAll(wsRoot, "https://", "wss://")
+	wsRoot = strings.ReplaceAll(wsRoot, "http://", "ws://")
+	wsRoot, _ = strings.CutSuffix(wsRoot, "/")
+
+	// Connect to ADB proxy WebSocket
+	//   wss://127.0.0.1:1443/devices/cvd-1/adb
+	// Since the operator is self-signed, skip verifying cert
+	dialer := websocket.Dialer {
+		TLSClientConfig: &tls.Config{
+			InsecureSkipVerify: true,
+		},
+	}
+	device := args[0]
+	wsUrl := fmt.Sprintf("%s/devices/%s/adb", wsRoot, device)
+	wsConn, _, err := dialer.Dial(wsUrl, nil)
+	if err != nil {
+		log.Fatal("Failed to connect ", wsUrl, ": ", err)
+		return err
+	}
+	defer wsConn.Close()
+
+	// Open local TCP port that ADB server connects to and make ADB server connect
+	// to the port
+	l, err := net.Listen("tcp", "127.0.0.1:0")
```

Comment:
Out of curiosity: Why port number is zero here?

Reply:
If use port number 0 while listening, it assigns random available port number while listening.

Reply:
Acknowledged. Thanks!